pySDC/implementations/problem_classes/Burgers.py

```diff
@@ -0,0 +1,315 @@
+import numpy as np
+
+from pySDC.implementations.datatype_classes.mesh import mesh, imex_mesh
+from pySDC.implementations.problem_classes.generic_spectral import GenericSpectralLinear
+
+
+class Burgers1D(GenericSpectralLinear):
+    """
+    See https://en.wikipedia.org/wiki/Burgers'_equation for the equation that is solved.
+    Discretization is done with a Chebychev method, which requires a first order derivative formulation.
+    Feel free to do a more efficient implementation using an ultraspherical method to avoid the first order business.
+
+    Parameters:
+       N (int): Spatial resolution
+       epsilon (float): viscosity
+       BCl (float): Value at left boundary
+       BCr (float): Value at right boundary
+       f (int): Frequency of the initial conditions
+       mode (str): 'T2U' or 'T2T'. Use 'T2U' to get sparse differentiation matrices
+    """
+
+    dtype_u = mesh
+    dtype_f = imex_mesh
+
+    def __init__(self, N=64, epsilon=0.1, BCl=1, BCr=-1, f=0, mode='T2U', **kwargs):
+        self._makeAttributeAndRegister(*locals().keys(), localVars=locals(), readOnly=True)
```

Comment:
Careful : `self` also belongs to `locals()` at this point and is added to the parameters of the object. If this is not wanted, maybe add a safety check in `_makeAttributeAndRegister` to ignore `self` in the names list ...